entsoe-cgmes-balances-adjustment/src/test/java/com/powsybl/entsoe/cgmes/balances_adjustment/util/CgmesBoundariesAreaTest.java

```diff
@@ -38,21 +34,36 @@ void testWithNoArea() {
     }
 
     @Test
-    void testWithArea() {
+    void testWithAreaFromOldExtension() {
         Network network = Network.read("controlArea.xiidm", getClass().getResourceAsStream("/controlArea.xiidm"));
-        NetworkAreaFactory factory = new CgmesBoundariesAreaFactory(new ArrayList<>(network.getExtension(CgmesControlAreas.class).getCgmesControlAreas()));
+        NetworkAreaFactory factory = new CgmesBoundariesAreaFactory(network.getAreaStream().toList());
         NetworkArea area = factory.create(network);
 
-        List<Double> ps = Stream.of(
-                    network.getDanglingLine("_78736387-5f60-4832-b3fe-d50daf81b0a6"),
-                    network.getDanglingLine("_17086487-56ba-4979-b8de-064025a6b4da"),
-                    network.getDanglingLine("_b18cd1aa-7808-49b9-a7cf-605eaf07b006"))
-                .map(dl -> dl.getBoundary().getP()).collect(Collectors.toList());
-        double sum = ps.stream().mapToDouble(n -> n).sum();
+        double sum = Stream.of(
+                        network.getDanglingLine("_78736387-5f60-4832-b3fe-d50daf81b0a6"),
+                        network.getDanglingLine("_17086487-56ba-4979-b8de-064025a6b4da"),
+                        network.getDanglingLine("_b18cd1aa-7808-49b9-a7cf-605eaf07b006"))
+                .mapToDouble(dl -> dl.getBoundary().getP()).sum();
 
         sum = sum + network.getTieLine("TL_fict").getDanglingLine1().getBoundary().getP();
         // FIXME: we miss the line "_b58bf21a-096a-4dae-9a01-3f03b60c24c7_fict_2" because it is not a tie line.
         assertEquals(-sum, area.getNetPosition(), DELTA_POWER);
         assertTrue(area.getContainedBusViewBuses().isEmpty());
     }
+
+    @Test
+    void testWithAreaWithoutAreaExtension() {
+        Network network = Network.read("controlAreaWithoutExtension.xml", getClass().getResourceAsStream("/controlAreaWithoutExtension.xml"));
+        NetworkAreaFactory factory = new CgmesBoundariesAreaFactory(network.getAreaStream().toList());
+        NetworkArea area = factory.create(network);
+
+        double sum = Stream.of(
+                        network.getDanglingLine("_78736387-5f60-4832-b3fe-d50daf81b0a6"),
+                        network.getDanglingLine("_17086487-56ba-4979-b8de-064025a6b4da"),
+                        network.getDanglingLine("_b18cd1aa-7808-49b9-a7cf-605eaf07b006"))
+                .mapToDouble(dl -> dl.getBoundary().getP()).sum();
+
+        assertEquals(-sum, area.getNetPosition(), DELTA_POWER);
+        assertTrue(area.getContainedBusViewBuses().isEmpty());
```

Comment:
Instead of duplicating the code, can you create a method taking the filename as parameter and call it in your two tests?